pkg/update/update.go

```diff
@@ -33,13 +33,55 @@ import (
 	"github.com/nitrictech/cli/pkg/view/tui"
 )
 
-func FetchLatestVersion() string {
-	latestVersionContents, err := os.ReadFile(cachePath())
+func FetchLatestCLIVersion() string {
+	return fetchLatestVersion("cli", cacheCLIPath())
+}
+
+func FetchLatestProviderVersion() string {
+	return fetchLatestVersion("nitric", cacheProviderPath())
+}
+
+func PrintOutdatedCLIWarning() {
+	currentVersion := strings.TrimPrefix(version.Version, "v")
+	latestVersion := FetchLatestCLIVersion()
+
+	// don't generate warning for non-production versions
+	if strings.Contains(currentVersion, "-") {
+		return
+	}
+
+	if currentVersion < latestVersion {
+		msg := fmt.Sprintf(`A new version of Nitric is available. To upgrade from version '%s' to '%s'`, currentVersion, latestVersion)
+		msg += ", visit https://nitric.io/docs/installation for instructions and release notes."
+
+		tui.Warning.Println(msg)
+	}
+}
+
+func PrintOutdatedProviderWarning(providerName string) {
+	latestVersion := FetchLatestProviderVersion()
+
+	var currentVersion string
+
+	providerParts := strings.SplitN(providerName, "@", 2)
+	if len(providerParts) == 2 {
+		providerName = providerParts[0]
+		currentVersion = providerParts[1]
+	}
+
+	if currentVersion < latestVersion {
```

Comment:
Should the output only work if the provider format was correct? Or exit earlier. Also, the currentVersion < latestVersion comparison will fail when either values get to double digits. Might need to split it into parts and compare each part numerically.
i.e. "1.4.2" < "1.10.2" will be false

Reply:
I think at this stage the provider has been validated and downloaded. Yeah it probably should.

Good catch with the string values, this would also be the case with the CLI version check we already do.